src/ActionMonitor/Monitors/PreviewMonitor.php

```diff
@@ -90,13 +661,17 @@ public function post_to_preview_instance( $post_ID, $post ) {
 		$original_post = get_post( $post->post_parent );
 
 		$this_is_a_publish_not_a_preview = 
-			$original_post && $original_post->post_modified === $post->post_modified;
+			$original_post
+			&& $original_post->post_modified === $post->post_modified
+			&& ! $is_gatsby_preview;
```

Comment:
@smthomas this check for ! $is_gatsby_preview didn't used to exist here which sometimes caused events to not be stored when pressing preview without updating content